packages/datadog_session_replay/android/src/main/kotlin/com/datadoghq/flutter/sessionreplay/feature/SegmentRequestFactory.kt

```
@@ -0,0 +1,91 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2025-Present Datadog, Inc.
+ */
+
+package com.datadoghq.flutter.sessionreplay.feature
+
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.net.Request
+import com.datadog.android.api.net.RequestExecutionContext
+import com.datadog.android.api.net.RequestFactory
+import com.datadog.android.api.storage.RawBatchEvent
+import java.util.Locale
+import java.util.UUID
+import okhttp3.RequestBody
+import okio.Buffer
+
+internal class SegmentRequestFactory(
+    private val customEndpointUrl: String?,
+    private val batchesToSegmentsMapper: BatchesToSegmentsMapper,
+    private val segmentRequestBodyFactory: SegmentRequestBodyFactory = SegmentRequestBodyFactory()
+) : RequestFactory {
+    override fun create(
+        context: DatadogContext,
+        executionContext: RequestExecutionContext,
+        batchData: List<RawBatchEvent>,
+        batchMetadata: ByteArray?
+    ): Request {
+        val serializedSegmentPair = batchesToSegmentsMapper.map(context, batchData.map { it.data })
+        if (serializedSegmentPair.isEmpty()) {
+            @Suppress("ThrowingInternalException")
+            throw InvalidPayloadFormatException(
```

why are we throwing an exception instead of logging it to telemetry?

---

It's caught further up the chain by the Core which reports it to Telemetry. This is the only `RequestFactory` that does this, but I think it's also the only one that has the extra check.